platform/android/export/export.cpp

```diff
@@ -2418,16 +2474,40 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 				store_in_gradle_project(img_path, data, Z_NO_COMPRESSION);
 			}
 		}
+	}
 
-        //TODO: replicate the functionality of _fix_manifest
+	Error gradle_build(const Ref<EditorExportPreset> &p_preset, bool p_debug, const String &p_path, String &src_apk, int p_flags = 0) {
 
-        //        String file = "res://android/build/AndroidManifest.xml";
-        //
-        //        Vector<uint8_t> android_manifest_file_data = FileAccess::get_file_as_array(file);
-        //        _fix_manifest(p_preset, android_manifest_file_data,  p_flags & (DEBUG_FLAG_DUMB_CLIENT | DEBUG_FLAG_REMOTE_DEBUG));
+		{ //test that installed build version is alright
+			FileAccessRef f = FileAccess::open("res://android/.build_version", FileAccess::READ);
+			if (!f) {
+				EditorNode::get_singleton()->show_warning(TTR("Trying to build from a custom built template, but no version info for it exists. Please reinstall from the 'Project' menu."));
+				return ERR_UNCONFIGURED;
+			}
+			String version = f->get_line().strip_edges();
+			if (version != VERSION_FULL_CONFIG) {
+				EditorNode::get_singleton()->show_warning(vformat(TTR("Android build version mismatch:\n   Template installed: %s\n   Godot Version: %s\nPlease reinstall Android build template from 'Project' menu."), version, VERSION_FULL_CONFIG));
+				return ERR_UNCONFIGURED;
+			}
+		}
+		//build project if custom build is enabled
+		String sdk_path = EDITOR_GET("export/android/custom_build_sdk_path");
+		ERR_FAIL_COND_V_MSG(sdk_path == "", ERR_UNCONFIGURED, "Android SDK path must be configured in Editor Settings at 'export/android/custom_build_sdk_path'.");
+
+		String output_gradle_path = p_path.replace(ProjectSettings::get_singleton()->get_resource_path(), "")
+											.replace(".apk", "")
+											.insert(0, "../../../../../..") +
+									"_${variant.name}.apk";
+
+		_update_custom_build_project(output_gradle_path); //alters the build.gradle, android manifest, etc.
+		_copy_icon_gradle(p_preset);
+		Error copy_value_xml_err = _copy_value_xml_files(p_preset, p_debug);
+
+		String manifest_path = "res://android/build/AndroidManifest.xml";
+		Error err = _fix_manifest_plaintext(p_preset, manifest_path, p_flags & (DEBUG_FLAG_DUMB_CLIENT | DEBUG_FLAG_REMOTE_DEBUG));
 
 		APKExportData ed;
-		Error err = export_project_files(p_preset, save_gradle_project_file, &ed, save_gradle_project_so);
+		err = export_project_files(p_preset, save_gradle_project_file, &ed, save_gradle_project_so);
```

Comment:
Do you mean to overwrite "err" without checking the result from line 2507 first?